Src/Particle/AMReX_NeighborList.H

```diff
@@ -108,9 +108,13 @@ public:
         m_bins.build(np_total, pstruct_ptr, bx,
                      [=] AMREX_GPU_HOST_DEVICE (const ParticleType& p) noexcept -> IntVect
                      {
-                         return IntVect(AMREX_D_DECL(static_cast<int>((p.pos(0)-plo[0])*dxi[0]) - lo.x,
-                                                     static_cast<int>((p.pos(1)-plo[1])*dxi[1]) - lo.y,
-                                                     static_cast<int>((p.pos(2)-plo[2])*dxi[2]) - lo.z));
+                         AMREX_D_DECL(AMREX_ASSERT((p.pos(0)-plo[0])*dxi[0] - lo.x),
```

Comment:
Missing `>= 0`?  Also it might be better to do `AMREX_D_TERM(AMREX_ASSERT();, AMREX_ASSERT();, AMREX_ASSERT());`.  `AMREX_D_DECL` always work, but it depends on the `,` has lower priority than `?:`.  Without looking up the rules, it's hard to be sure.

Reply:
i defer to weiqun as always :-)

Reply:
You are right, fixed.